sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -4600,3 +4600,30 @@ case class ArrayExcept(left: Expression, right: Expression) extends ArrayBinaryL
   override protected def withNewChildrenInternal(
     newLeft: Expression, newRight: Expression): ArrayExcept = copy(left = newLeft, right = newRight)
 }
+
+@ExpressionDescription(
+  usage = "_FUNC_(array) - Removes null values from the array.",
+  examples = """
+    Examples:
+      > SELECT _FUNC_(array(1, 2, 3, null));
+       [1,2,3]
+  """,
+  group = "array_funcs",
+  since = "3.4.0")
+case class ArrayCompact(child: Expression)
+  extends RuntimeReplaceable with UnaryLike[Expression] with ExpectsInputTypes with NullIntolerant {
```

Comment:
We should remove `UnaryLike`, `ExpectsInputTypes` and `NullIntolerant` here and implements `InheritAnalysisRules`.

Reply:
Please try to use this method.
```
    lazy val ArrayType(elementType, nullable) = child.dataType
    lazy val lv = NamedLambdaVariable("arg", elementType, nullable)
    lazy val function = f(lv)
    lazy val lambda = LambdaFunction(function, Seq(lv))
```

Reply:
I tried this this will give `StackOverflowError` exception.

Reason: 
`child` is overridden method which in turn calls `replacement` 
```
Code: 
trait InheritAnalysisRules extends UnaryLike[Expression] { self: RuntimeReplaceable =>
  override def child: Expression = replacement
```

If I am trying to use `expr.dataType` this gives `UnresolvedException` which is correct cz `replacement` is not yet resolved

Reply:
You can refer https://github.com/apache/spark/blob/d2212e352043063b14cac8ec53fe65731ba593f5/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala#L2500